### PR TITLE
Bug 1221377 - Preventing keyboard when an actionmenu is being shown r=etienne

### PR DESCRIPTION
**Pull request by albertopq**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=4c6fd60c833c218add36ec2986ef2d9285952e0a

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=d7db6670ed578d38c94b2717b0237dd19766e228

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=2eb4e083be27b3878d279370b63be0da8b792335

File: apps/system/js/input_window_manager.js [hunk 1]
```diff
@@ -226,17 +228,31 @@
...
+      // an actionmenu/activity/attention. These are the event
+      // pairs:
+      // actionmenuopening - actionmenuopen
+      // activityrequesting - activityopening
+      // attentionopening - attentionopened
+      case 'actionmenuopening':
       case 'activityrequesting':
-      case 'activityopening':
+      case 'attentionopening':
+        this.preventInput = true;
```
Comment (timdream) [inline apps/system/js/input_window_manager.js]: `preventInput` -> `blockInput`.

File: apps/system/js/input_window_manager.js [hunk 2]
```diff
@@ -226,17 +228,31 @@
...
       case 'attentionrecovering':
-      case 'attentionopening':
+      case 'activityopening':
       case 'attentionclosing':
       case 'attentionopened':
       case 'attentionclosed':
       case 'notification-clicked':
       case 'applicationsetupdialogshow':
+        this.preventInput = false;
         this.hideInputWindowImmediately();
```
Comment (timdream) [inline apps/system/js/input_window_manager.js]: Do `navigator.mozInputMethod.removeFocus();` here just to be safe.

File: apps/system/js/input_window_manager.js [hunk 3]
```diff
@@ -402,6 +418,10 @@
 
   InputWindowManager.prototype.showInputWindow =
   function iwm_showInputWindow(layout) {
+    if (this.preventInput) {
+      return;
```
Comment (timdream) [inline apps/system/js/input_window_manager.js]: And here too `navigator.mozInputMethod.removeFocus();`

File: apps/system/js/input_window_manager.js (hunk 2 above)
Comment (albertopq) [inline apps/system/js/input_window_manager.js]: Is already done in the next line, isn't it?

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=11da2b40cb0f6798f147473fa20b7a42963933f0

File: apps/system/js/input_window_manager.js [hunk 4]
```diff
@@ -226,17 +228,31 @@
           KeyboardManager._onKeyboardKilled(manifestURL);
         }
         break;
+
+      // We want to avoid the keyboard showing while opening
+      // an actionmenu/activity/attention. These are the event
+      // pairs:
+      // actionmenuopening - actionmenuopen
+      // activityrequesting - activityopening
```
Comment (etiennesegonzac) [inline apps/system/js/input_window_manager.js]: what happens if we cancel the activity from the picker? We won't get the `activityopening` event...

Comment (albertopq) [inline apps/system/js/input_window_manager.js]: In that case, the  'actionmenuopen', is already canceling the blockInput flag.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=869f244689559ef9b07acee25d8c9403d0d22a2a

**Pull request merged**